src/Orleans/Serialization/SerializationContext.cs

```diff
@@ -60,7 +62,7 @@ public override bool Equals(object x, object y)
 
             public override int GetHashCode(object obj)
             {
-                return obj == null ? 0 : obj.GetHashCode();
+                return obj == null ? 0 : RuntimeHelpers.GetHashCode(obj);
```

Comment:
Not strictly required for this change, but a potential bug. Just as we use `object.ReferenceEquals` above this, we should call `RuntimeHelpers.GetHashCode` rather than `obj.GetHashCode` so that we are not dependent on user-defined implementations of GetHashCode. It's not likely to cause an issue, but it's possible.